src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Ssl.cs

```diff
@@ -59,9 +59,28 @@ internal enum PAL_TlsIo
             Renegotiate,
         }
 
+        internal enum PAL_NwStatusUpdates
+        {
+            UnknownError = 0,
+            FramerStart = 1,
+            HandshakeFinished = 2,
+            HandshakeFailed = 3,
+            ConnectionReadFinished = 4,
+            ConnectionWriteFinished = 5,
+            ConnectionWriteFailed = 6,
+            ConnectionError = 7,
+            ConnectionCancelled = 8,
+        }
+
         [LibraryImport(Interop.Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_SslCreateContext")]
         internal static partial System.Net.SafeSslHandle SslCreateContext(int isServer);
 
+        [LibraryImport(Interop.Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_NwCreateContext")]
+        internal static partial System.Net.SafeSslHandle NwCreateContext(int isServer);
+
+        [LibraryImport(Interop.Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_NwInit")]
+        internal static unsafe partial int NwInit(delegate* unmanaged<IntPtr, PAL_NwStatusUpdates, IntPtr, IntPtr, int> statusCallback, delegate* unmanaged<IntPtr, byte*, void**, int> readCallback, delegate* unmanaged<IntPtr, byte*, void**, int> writeCallback);
```

Comment:
This would be helpful tho, especially for the possible future implementations/experiments. (QUIC API is there and can be used with couple of more interop additions to this. (there are some bugs tho))